# **The FI-ndings**
## InmoPredict - Limpieza de datos

**Objetivo del notebook:** preparar un dataset limpio y validado para estimar el valor de renta de la propiedad, utilizando únicamente variables observables de la vivienda y su contexto territorial.




# **1. Importación de librerías**







Se importan las librerías necesarias para realizar la limpieza de los datos.

In [ ]:
import pandas as pd
import numpy as np

print('Librerías cargadas correctamente')

Librerías cargadas correctamente


# **2. Carga de archivo viviendas.csv**

En esta sección se carga el archivo principal del proyecto, viviendas.csv, desde Google Drive, ya que es la forma más práctica de trabajar en Google Colab.

In [ ]:
# Conexión con Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Lectura de archivo viviendas.csv

# Colocar la ruta según la ubicación real del archivo en Google Drive
ruta_viviendas = '/content/drive/MyDrive/viviendas.csv'

# Cargar archivo
df_viviendas = pd.read_csv(ruta_viviendas)

print('Archivo viviendas.csv cargado correctamente')

Archivo viviendas.csv cargado correctamente


/tmp/ipykernel_8211/1089436112.py:7: DtypeWarning: Columns (1,4,27,44) have mixed types. Specify dtype option on import or set low_memory=False.
  df_viviendas = pd.read_csv(ruta_viviendas)


# **3. Exploración inicial**
En esta etapa se revisa la estructura general del dataset antes de realizar cualquier limpieza o transformación.

In [ ]:
# Vistazo de las primeras 5 filas para verificar que la información se haya cargado correctamente.
df_viviendas.head()

,folioviv,tipo_viv,mat_pared,mat_techos,mat_pisos,antiguedad,antigua_ne,cocina,cocina_dor,cuart_dorm,...,tot_hom,tot_muj,tot_hog,ubica_geo,tam_loc,est_socio,est_dis,upm,factor,procaptar
0,100001901,7,8,10,3,30,NaN,1,2.0,1,...,2,2,1,1001,1,3,1,1,207,0
1,100001902,1,8,10,3,30,NaN,1,2.0,3,...,2,2,1,1001,1,3,1,1,207,0
2,100001904,1,8,10,3,,1.0,1,2.0,2,...,1,1,1,1001,1,3,1,1,207,0
3,100001905,1,8,10,3,25,NaN,1,2.0,2,...,3,1,1,1001,1,3,1,1,207,0
4,100002501,4,8,10,3,9,NaN,1,2.0,2,...,1,3,1,1001,1,2,2,2,196,0


In [ ]:
# Se indica el número de registros por número de columnas
print('Número de filas y columnas:', df_viviendas.shape)

Número de filas y columnas: (90324, 82)


In [ ]:
# Se muestra el listado completo de las variables disponibles
print('\nColumnas disponibles:')
print(df_viviendas.columns)


Columnas disponibles:
Index(['folioviv', 'tipo_viv', 'mat_pared', 'mat_techos', 'mat_pisos',
       'antiguedad', 'antigua_ne', 'cocina', 'cocina_dor', 'cuart_dorm',
       'num_cuarto', 'lugar_coc', 'agua_ent', 'ab_agua', 'agua_noe',
       'dotac_agua', 'excusado', 'uso_compar', 'sanit_agua', 'biodigest',
       'bano_comp', 'bano_excus', 'bano_regad', 'drenaje', 'disp_elect',
       'focos', 'focos_ahor', 'combus', 'fogon_chi', 'eli_basura', 'tenencia',
       'renta', 'estim_pago', 'pago_viv', 'pago_mesp', 'tipo_adqui',
       'viv_usada', 'finan_1', 'finan_2', 'finan_3', 'finan_4', 'finan_5',
       'finan_6', 'finan_7', 'finan_8', 'num_dueno1', 'hog_dueno1',
       'num_dueno2', 'hog_dueno2', 'escrituras', 'lavadero', 'fregadero',
       'regadera', 'tinaco_azo', 'cisterna', 'pileta', 'calent_sol',
       'calent_gas', 'calen_lena', 'medid_luz', 'bomba_agua', 'tanque_gas',
       'aire_acond', 'calefacc', 'p_grietas', 'p_pandeos', 'p_levanta',
       'p_humedad', 'p_fractura', '

In [ ]:
# Se revisa el tipo de dato de cada columna
print('\nTipos de datos:')
df_viviendas.dtypes


Tipos de datos:


,0
folioviv,int64
tipo_viv,object
mat_pared,int64
mat_techos,int64
mat_pisos,object
...,...
est_socio,int64
est_dis,int64
upm,int64
factor,int64


# **4. Selección de variables útiles para el modelo**

En esta sección se seleccionan únicamente las variables que serán utilizadas para el análisis y modelado de InmoPredict.

In [ ]:
# Se conservan únicamente las variables previamente agrupadas.

variables_modelo = [
    # Identificación, filtro y variable objetivo
    'folioviv',
    'tenencia',
    'renta',

    # Ubicación y contexto territorial
    'ubica_geo',
    'tam_loc',
    'est_socio',

    # Características físicas de la vivienda
    'tipo_viv',
    'mat_pared',
    'mat_techos',
    'mat_pisos',
    'antiguedad',
    'antigua_ne',
    'cocina',
    'cuart_dorm',
    'num_cuarto',
    'lugar_coc',

    # Servicios básicos y sanitarios
    'agua_ent',
    'ab_agua',
    'dotac_agua',
    'excusado',
    'uso_compar',
    'sanit_agua',
    'bano_comp',
    'bano_excus',
    'bano_regad',
    'drenaje',
    'disp_elect',
    'focos',
    'combus',
    'eli_basura',

    # Amenidades y equipamiento de la propiedad
    'lavadero',
    'fregadero',
    'regadera',
    'tinaco_azo',
    'cisterna',
    'pileta',
    'calent_sol',
    'calent_gas',
    'medid_luz',
    'bomba_agua',
    'tanque_gas',
    'aire_acond',
    'calefacc',

    # Problemas físicos o deterioro
    'p_grietas',
    'p_pandeos',
    'p_levanta',
    'p_humedad',
    'p_fractura',
    'p_electric',
    'p_tuberias'
]

df_modelo = df_viviendas[variables_modelo].copy()

print('Dataset reducido a variables útiles para el proyecto')
print('Número de filas y columnas:', df_modelo.shape)

Dataset reducido a variables útiles para el proyecto
Número de filas y columnas: (90324, 50)


In [ ]:
# Se muestra el listado completo de las variables seleccionadas
print('\nColumnas disponibles:')
print(df_modelo.columns)


Columnas disponibles:
Index(['folioviv', 'tenencia', 'renta', 'ubica_geo', 'tam_loc', 'est_socio',
       'tipo_viv', 'mat_pared', 'mat_techos', 'mat_pisos', 'antiguedad',
       'antigua_ne', 'cocina', 'cuart_dorm', 'num_cuarto', 'lugar_coc',
       'agua_ent', 'ab_agua', 'dotac_agua', 'excusado', 'uso_compar',
       'sanit_agua', 'bano_comp', 'bano_excus', 'bano_regad', 'drenaje',
       'disp_elect', 'focos', 'combus', 'eli_basura', 'lavadero', 'fregadero',
       'regadera', 'tinaco_azo', 'cisterna', 'pileta', 'calent_sol',
       'calent_gas', 'medid_luz', 'bomba_agua', 'tanque_gas', 'aire_acond',
       'calefacc', 'p_grietas', 'p_pandeos', 'p_levanta', 'p_humedad',
       'p_fractura', 'p_electric', 'p_tuberias'],
      dtype='object')


In [ ]:
# Se revisa el tipo de dato de cada columna
print('\nTipos de datos:')
df_modelo.dtypes


Tipos de datos:


,0
folioviv,int64
tenencia,int64
renta,object
ubica_geo,int64
tam_loc,int64
est_socio,int64
tipo_viv,object
mat_pared,int64
mat_techos,int64
mat_pisos,object


# **5. Revisión y tratamiento de tipo de dato**
El objetivo de esta sección es transformar las variables a formato númerico.

In [ ]:
# Transformación a tipo de formato númerico
columnas_a_convertir = [
    col for col in df_modelo.columns
]

for columna in columnas_a_convertir:
    df_modelo[columna] = pd.to_numeric(
        df_modelo[columna],
        errors='coerce'
    )

print('Conversión a formato numérico realizada correctamente')


Conversión a formato numérico realizada correctamente


In [ ]:
# Se revisa el tipo de dato de cada columna
print('\nTipos de datos:')
df_modelo.dtypes


Tipos de datos:


,0
folioviv,int64
tenencia,int64
renta,float64
ubica_geo,int64
tam_loc,int64
est_socio,int64
tipo_viv,float64
mat_pared,int64
mat_techos,int64
mat_pisos,float64


Después de convertir las variables a formato numérico, se transforman a tipo entero.

In [ ]:
# Se convierten las variables numéricas a enteros.
for columna in columnas_a_convertir:
    df_modelo[columna] = df_modelo[columna].astype('Int64')

print('Variables convertidas a tipo entero correctamente')

Variables convertidas a tipo entero correctamente


In [ ]:
# Se revisa el tipo de dato de cada columna
print('\nTipos de datos:')
df_modelo.dtypes


Tipos de datos:


,0
folioviv,Int64
tenencia,Int64
renta,Int64
ubica_geo,Int64
tam_loc,Int64
est_socio,Int64
tipo_viv,Int64
mat_pared,Int64
mat_techos,Int64
mat_pisos,Int64


# **6. Revisión y tratamiento de valores nulos y duplicados**
El objetivo de esta sección es identificar valores faltantes y duplicados únicamente en las variables que serán utilizadas para estimar el valor de renta de la propiedad.

In [ ]:
# Se revisan registros duplicados exactos y duplicados por identificador de vivienda.

duplicados_exactos = df_modelo.duplicated().sum()
duplicados_folioviv = df_modelo['folioviv'].duplicated().sum()

print('Duplicados exactos:', duplicados_exactos)
print('Duplicados por folioviv:', duplicados_folioviv)

Duplicados exactos: 0
Duplicados por folioviv: 0


In [ ]:
# Variables con valores nulos

# Se filtran únicamente las variables que presentan valores nulos.
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

,0
renta,78758
tipo_viv,85
mat_pisos,1
antiguedad,12278
antigua_ne,78046
ab_agua,4175
dotac_agua,4175
uso_compar,1498
sanit_agua,1498
bano_comp,1499


Se identificó que las variables con nulos, no son errores, si no que son nulos por la lógica del cuestionario, ya que corresponden principalmente a preguntas que solo se contestan si aplica.

Por esta razón, los registros no serán eliminados. En su lugar, se realizará un tratamiento de los datos para conservar la información y preparar el dataset para el modelado.

# Variable renta

Indica el monto de la renta mensual reportada.

Se identificó que si existen valores nulos correspondientes a las viviendas que no entrarn en la etiqueta de *"rentada"*

In [ ]:
# Comprobar si los valores nulos corresponden a viviendas que son rentadas
print('Total de registros que no son rentadas: ', (df_modelo["tenencia"]!=1).sum())

Total de registros que no son rentadas:  78758


Se encontraron 78,758 valores correspondientes a las viviendas que no son rentadas, los cuales coinciden con los casos valores nulos en la variable renta

Para conservar estos registros se remplazaron los valore nulos por el valor `0`, que será interpretado como *"No aplica"*

In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["renta"] = df_modelo["renta"].fillna(0)

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulos en renta
print('Total de registros nulos en renta: ', (df_modelo['renta'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

Total de registros nulos en renta:  0


Registros con valores nulos:


,0
tipo_viv,85
mat_pisos,1
antiguedad,12278
antigua_ne,78046
ab_agua,4175
dotac_agua,4175
uso_compar,1498
sanit_agua,1498
bano_comp,1499
bano_excus,1498


# Variable tipo_viv

Indica la clase de vivienda particular.

Se identificó que si existen valores nulos correspondientes a las viviendas que presentaron un dato no especificado.

Se encontraron 85 valores correspondientes a las viviendas que no especificaron el tipo de vivienda.

Para conservar estos registros se remplazaron los valores nulos por el valor 0, que será interpretado como "No especificado"


In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["tipo_viv"] = df_modelo["tipo_viv"].fillna(0)


Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulos en tipo_viv
print('Total de registros nulos en tipo_viv: ', (df_modelo['tipo_viv'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]


Total de registros nulos en tipo_viv:  0


Registros con valores nulos:


,0
mat_pisos,1
antiguedad,12278
antigua_ne,78046
ab_agua,4175
dotac_agua,4175
uso_compar,1498
sanit_agua,1498
bano_comp,1499
bano_excus,1498
bano_regad,1498


# Variable mat_pisos

Indica el material predominante en el piso de la vivienda.

Se identificó que si existen valores nulos correspondientes a las viviendas que presentaron un dato no especificado en el material del piso.

Se encontró 1 valor correspondiente a la vivienda que no especificó el tipo de material del suelo.

Para conservar este registro se rempló este valor nulo por el valor 0, que será interpretado como "No especificado"


In [ ]:
# Se reemplazó el valor nulo por 0.
df_modelo["mat_pisos"] = df_modelo["mat_pisos"].fillna(0)


Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulo en mat_pisos
print('Total de registros nulos en mat_pisos: ', (df_modelo['mat_pisos'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]


Total de registros nulos en mat_pisos:  0


Registros con valores nulos:


,0
antiguedad,12278
antigua_ne,78046
ab_agua,4175
dotac_agua,4175
uso_compar,1498
sanit_agua,1498
bano_comp,1499
bano_excus,1498
bano_regad,1498
focos,475


# Variable antigua_ne


Indica si la antigüedad de la vivienda no fue especificada.

Se identificó que si existen valores no nulos en antigua_ne, deberia de haber en antiguedad. Sin embargo, por algun motivo no se identifican como nulos.



In [ ]:
# Para el tratamiento de antiguedad Y antigua_ne
# Se crea una variable bandera y se convierte antiguedad a numérica.

# Crear bandera:
# 1 = No se especificó la antigüedad
# 0 = Sí se especificó la antigüedad

df_modelo['antiguedad_no_especificada'] = np.where(
    df_modelo['antigua_ne'].notna(),
    1,
    0
)

# Limpiar posibles espacios o códigos especiales en antiguedad
df_modelo['antiguedad'] = (
    df_modelo['antiguedad']
    .astype(str)
    .str.strip()
    .replace(['', ' ', '&', 'N', 'nan', 'NaN', '<NA>'], np.nan)
)

# Convertir antiguedad a número
df_modelo['antiguedad'] = pd.to_numeric(
    df_modelo['antiguedad'],
    errors='coerce'
)

# Revisar nulos reales después de la conversión
print('Nulos reales en antiguedad después de limpiar:')
print(df_modelo['antiguedad'].isna().sum())

print('\nCasos con antigüedad no especificada:')
print(df_modelo['antiguedad_no_especificada'].sum())

Nulos reales en antiguedad después de limpiar:
12278

Casos con antigüedad no especificada:
12278


Después de limpiar y convertir `antiguedad` a formato numérico, se encontraron 12,278 valores faltantes, los cuales coinciden con los casos registrados en `antigua_ne`.

Para conservar estos registros se asignó el valor `-1` a los casos no especificados. Este valor se utiliza como código interno, ya que no pertenece al rango válido de antigüedad de la vivienda.

Adicionalmente, se creó la variable `antiguedad_no_especificada`, donde `1` indica que la antigüedad no fue reportada y `0` indica que sí fue especificada.

Despues se eliminó `antigua_ne`, debido a que su información quedó representada en la nueva variable bandera.


In [ ]:
# Reemplazar nulos de antiguedad con -1
df_modelo['antiguedad'] = df_modelo['antiguedad'].fillna(-1)

# Eliminar antigua_ne porque ya fue transformada en una variable útil
df_modelo = df_modelo.drop(columns=['antigua_ne'])

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
# Validaciones
print('Validaciones')

# Verificar si la variable antigua_ne fue eliminada
print('\n¿La columna antigua_ne sigue en el dataset?', 'antigua_ne' in df_modelo.columns)

# Verificar nulos en antiguedad
print('\nNulos en antiguedad:', df_modelo['antiguedad'].isna().sum())

# Verificar cantidad de valores -1
print('\nCasos con antiguedad = -1:', (df_modelo['antiguedad'] == -1).sum())

# Verificar cantidad de casos marcados como no especificados
print('\nCasos con antiguedad_no_especificada = 1:', df_modelo['antiguedad_no_especificada'].sum())

# Verificar valores únicos de la bandera
print('\nValores únicos en antiguedad_no_especificada:', df_modelo['antiguedad_no_especificada'].unique())

# Verificar rango de antiguedad
print('\nValor mínimo y máximo de antiguedad:', df_modelo['antiguedad'].min(), ',', df_modelo['antiguedad'].max())

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos: ')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

Validaciones

¿La columna antigua_ne sigue en el dataset? False

Nulos en antiguedad: 0

Casos con antiguedad = -1: 12278

Casos con antiguedad_no_especificada = 1: 12278

Valores únicos en antiguedad_no_especificada: [0 1]

Valor mínimo y máximo de antiguedad: -1.0 , 99.0


Registros con valores nulos: 


,0
ab_agua,4175
dotac_agua,4175
uso_compar,1498
sanit_agua,1498
bano_comp,1499
bano_excus,1498
bano_regad,1498
focos,475
combus,8


# Variable ab_agua y dotac_agua

La variable `ab_agua` indica el lugar de donde proviene el agua, mientras que `dotac_agua` indica la frecuencia con la que llega a la vivienda el agua entubada.

Ambas variables están relacionadas con `agua_ent` que indica el lugar en donde se encuentra ubicada la llave o mangera con la que se obtiene el agua.

Los nulos aparecen porque la pregunta solo aplica cuando la vivienda tiene agua entubada. Si `agua_ent` = 3, significa que la vivienda no tiene agua entubada, por lo tanto `ab_agua` y `dotac_agua`no aplican.


In [ ]:
# Comprobar si los valores nulos de ab_agua y dotac_agua coinciden con las viviendas sin agua entubada.
print('Total de registros que no tienen agua entubada: ', (df_modelo["agua_ent"]==3).sum())

Total de registros que no tienen agua entubada:  4175


Se encontraron 4,175 valores correspondientes a las viviendas que no cuentan con agua entubada, los cuales coinciden con los casos en donde no aplican en `ab_agua y dotac_agua`.

Para conservar estos registros se remplazaron los valore nulos por el valor `0`, que será interpretado como *"No aplica"*

In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["ab_agua"] = df_modelo["ab_agua"].fillna(0)
df_modelo["dotac_agua"] = df_modelo["dotac_agua"].fillna(0)

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.



In [ ]:
#Verificaciones

# Verificar nulos en ab_agua
print('Total de registros nulos en ab_agua: ', (df_modelo['ab_agua'].isnull().sum()))

# Verificar nulos en dotac_agua
print('\nTotal de registros nulos en dotac_agua: ', (df_modelo['dotac_agua'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

Total de registros nulos en ab_agua:  0

Total de registros nulos en dotac_agua:  0


Registros con valores nulos:


,0
uso_compar,1498
sanit_agua,1498
bano_comp,1499
bano_excus,1498
bano_regad,1498
focos,475
combus,8


# Variable uso_compar, sanit_agua, bano_comp, bano_excus y bano_regad

La variable `uso_compar` indica el uso compartido de la instalación sanitaria con los habitantes de otra vivienda, mientras que `sanit_agua` indica el funcionamiento de la instalación sanitaria con
o sin conexión de agua. **FALTA PONER DE QUE VAN bano_comp, bano_excus y bano_regad**

Ambas variables están relacionadas con `excusado` que indica si el lugar cuenta con instalación sanitaria o excusado en la vivienda para el desalojo de los desechos humanos.

Los nulos aparecen porque la pregunta solo aplica cuando la vivienda tiene intalación sanitaria o excusado. Si `escusado` = 3, significa que la vivienda no cuenta con dicho servicio, por lo tanto `uso_compar` y `sanit_agua` no aplican.

In [ ]:
# Comprobar si los valores nulos de uso_compar y sanit_agua coinciden con las viviendas sin instalación sanitaria o excusado.
print('Total de registros que no cuentan con instalación sanitaria o excusado: ', (df_modelo["excusado"]==3).sum())

Total de registros que no cuentan con instalación sanitaria o excusado:  1498


Se encontraron aproximadamente 1,498 valores correspondientes a las viviendas que no cuentan con instalación sanitaria o excusado, los cuales coinciden con los casos en donde no aplican en `uso_compar` y `sanit_agua`.

Para conservar estos registros se remplazaron los valore nulos por el valor `0`, que será interpretado como *"No aplica"*

In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["uso_compar"] = df_modelo["uso_compar"].fillna(0)
df_modelo["sanit_agua"] = df_modelo["sanit_agua"].fillna(0)
df_modelo["bano_comp"] = df_modelo["bano_comp"].fillna(0)
df_modelo["bano_excus"] = df_modelo["bano_excus"].fillna(0)
df_modelo["bano_regad"] = df_modelo["bano_regad"].fillna(0)

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulos en uso_compar
print('Total de registros nulos en uso_compar: ', (df_modelo['uso_compar'].isnull().sum()))

# Verificar nulos en sanit_agua
print('Total de registros nulos en sanit_agua: ', (df_modelo['sanit_agua'].isnull().sum()))

# Verificar nulos en bano_comp
print('Total de registros nulos en bano_comp: ', (df_modelo['bano_comp'].isnull().sum()))

# Verificar nulos en bano_excus
print('Total de registros nulos en bano_excus: ', (df_modelo['bano_excus'].isnull().sum()))

# Verificar nulos en bano_regad
print('Total de registros nulos en bano_regad: ', (df_modelo['bano_regad'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

Total de registros nulos en uso_compar:  0
Total de registros nulos en sanit_agua:  0
Total de registros nulos en bano_comp:  0
Total de registros nulos en bano_excus:  0
Total de registros nulos en bano_regad:  0


Registros con valores nulos:


,0
focos,475
combus,8


# Variable focos

Indica el total de focos que se usan para iluminar la vivienda, tanto en el interior como en el exterior.

Para conservar estos registros se remplazaron los valores nulos por el valor 0, que será interpretado como "No especificado"


In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["focos"] = df_modelo["focos"].fillna(0)

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulos en focos
print('Total de registros nulos en focos: ', (df_modelo['focos'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]



Total de registros nulos en focos:  0


Registros con valores nulos:


,0
combus,8


# Variable combus

Indica el combustible más utilizado para preparar o calentar alimentos en la vivienda.


Para conservar estos registros se remplazaron los valores nulos por el valor 0, que será interpretado como "No especificado"

In [ ]:
# Se reemplazan los valores nulos por 0.
df_modelo["combus"] = df_modelo["combus"].fillna(0)

Finalmente, se realizaron las validaciones correspondientes para asegurarse que el proceso se haya realizado correctamente.

In [ ]:
#Verificaciones

# Verificar nulos en combus
print('Total de registros nulos en combus: ', (df_modelo['combus'].isnull().sum()))

# Se filtran únicamente las variables que presentan valores nulos.
print('\n\nRegistros con valores nulos:')
df_modelo.isnull().sum()[df_modelo.isnull().sum() > 0]

Total de registros nulos en combus:  0


Registros con valores nulos:


,0


# **8. Exportar dataset limpio**
En esta sección se guarda el dataset limpio en formato CSV para usarlo posteriormente en el análisis exploratorio y modelado.

In [ ]:
df_modelo.to_csv('z', index=False)